tests/IntMapValidity.hs

```diff
@@ -60,4 +60,6 @@ maskRespected t =
     Tip _ _ -> True
     Bin _ binMask l r ->
       all (\x -> zero x binMask) (keys l) &&
-      all (\x -> not (zero x binMask)) (keys r)
+      all (\x -> not (zero x binMask)) (keys r) &&
+      maskRespected l &&
+      maskRespected r
```

Comment:
This implementation of `maskRespected` is at least quadratic. Can't you get it down to linear?

Reply:
That's something we could do, but I'm not sure if it would improve or hurt the readability. It would require fewer traversals of the tree but may make it more difficult to see how each invariant is tested.